pkg/provisioning/service.go

```diff
@@ -43,40 +48,79 @@ func NewService(
 	plService PipelineService,
 	connService ConnectorService,
 	procService ProcessorService,
+	pipelinesDir string,
 ) *Service {
 	return &Service{
 		db:               db,
 		logger:           logger.WithComponent("provisioning.Service"),
 		pipelineService:  plService,
 		connectorService: connService,
 		processorService: procService,
+		pipelinesPath:    pipelinesDir,
 	}
 }
 
-func (s *Service) ProvisionConfigFile(ctx context.Context, path string) ([]string, error) {
+func (s *Service) Init(ctx context.Context) error {
+	s.logger.Debug(ctx).Msg("initializing the provisioning service")
+
+	var multierr error
+	var files []string
+	err := filepath.WalkDir(s.pipelinesPath, func(path string, fileInfo fs.DirEntry, err error) error {
+		if strings.HasSuffix(path, ".yml") {
+			files = append(files, path)
+		}
+		return nil
+	})
+	if err != nil {
+		multierr = multierror.Append(multierr, cerrors.Errorf("could not iterate through the pipelines folder %q: %w", s.pipelinesPath, err))
+		return multierr
+	}
+
+	if len(files) == 0 {
+		s.logger.Warn(ctx).Str("pipelinesDir", s.pipelinesPath).Msg("configuration folder has no YAML files, no pipelines created")
```

Comment:
Maybe we can `return` at this point?

Reply:
if we return here, then we won't delete old pipelines